libcodechecker/analyze.py

```diff
@@ -0,0 +1,290 @@
+                                    "to BE PROHIBITED from use in the "
+                                    "analysis.")
+
+    checkers_opts.add_argument('--enable-all',
+                               dest="enable_all",
+                               action='store_true',
+                               required=False,
+                               default=argparse.SUPPRESS,
+                               help="Force the running analyzers to use "
+                                    "almost every checker available. The "
+                                    "checker groups 'alpha.', 'debug.' and "
+                                    "'osx.' (on Linux) are NOT enabled "
+                                    "automatically and must be EXPLICITLY "
+                                    "specified. WARNING! Enabling all "
+                                    "checkers might result in the analysis "
+                                    "losing precision and stability, and "
+                                    "could even result in a total failure of "
+                                    "the analysis. USE WISELY AND AT YOUR "
+                                    "OWN RISK!")
+
+    add_verbose_arguments(parser)
+    parser.set_defaults(func=main)
+
+
+def main(args):
+    """
+    Perform analysis on the given logfiles and store the results in a machine-
+    readable format.
+    """
+
+    context = generic_package_context.get_context()
+
+    # Parse the JSON CCDBs and retrieve the compile commands.
+    actions = []
+
+    for log_file in args.logfile:
+        if not os.path.exists(log_file):
+            LOG.error("The specified logfile '" + log_file + "' does not "
+                      "exist!")
+            continue
+
+        actions += log_parser.parse_log(log_file,
+                                        args.add_compiler_defaults)
+
+    if 'enable_all' in args:
+        # enable_all is handled in
+        # libcodechecker/analyze/analyzers/analyzer_types.py.
+        LOG.info("'--enable-all' was supplied for this analysis.")
+
+    # Run the analysis.
+    args.output_path = os.path.abspath(args.output_path)
+    if os.path.isdir(args.output_path):
+        shutil.rmtree(args.output_path)
+    os.makedirs(args.output_path)
+
+    LOG.debug("Output will be stored to: '" + args.output_path + "'")
+
+    metadata = {'action_num': len(actions),
+                'command': sys.argv,
+                'versions': {
+                    'codechecker': "{0} ({1})".format(context.package_git_tag,
+                                                      context.package_git_hash)
+                },
+                'working_directory': os.getcwd(),
+                'output_path': args.output_path}
+
+    if 'name' in args:
+        metadata['name'] = args.name
+
+    if 'skipfile' in args:
+        # Skip information needs to be saved because reports in a header
+        # can only be suppressed by the report-server used in 'store' later
```

Comment:
Here the skip and the suppress functionality is mixed in the comment. Skip will not store the results, with suppress the results will be stored but suppressed.